include/clients/nrt/MLPClassifierClient.hpp

```diff
@@ -251,7 +267,20 @@ class MLPClassifierClient : public FluidBaseClient,
 
 private:
   FluidInputTrigger                         mTrigger;
-  ParameterTrackChanges<IndexVector, index> mTracker;
+  ParameterTrackChanges<index, index, IndexVector, index> mTracker;
+
+  void updateParameters(){
+    const index nLayers = mAlgorithm.size();
+    if(nLayers > 1){
```

Comment:
Just a question about this condition: presumably it's always operatively true, but what's the upshot if read / load is called and this isn't true? Doesn't that mean that the host and algorithm fall out of sync again?

Reply:
I guess the hidden layers parameter could be set to an empty vector and the activation to default, probably never a good outcome anyway, but at least consistent